RealmSwift/Tests/MigrationTests.swift

```diff
@@ -51,7 +51,9 @@ class MigrationTests: TestCase {
     }
 
     // migrate realm at path and ensure migration
-    private func migrateAndTestRealm(realmPath: String, shouldRun: Bool = true, schemaVersion: UInt = 1, autoMigration: Bool = false, block: MigrationBlock? = nil) {
+    private func migrateAndTestRealm(realmPath: String, isNecessary: Bool = false, shouldRun: Bool = true, schemaVersion: UInt = 1, autoMigration: Bool = false, block: MigrationBlock? = nil) {
+        var error: NSError? = nil
+        XCTAssertEqual(Realm.migrationRequired(atPath: realmPath, error: &error), isNecessary)
```

Comment:
1. that's not always the case since you'll get all the errors that could happen when opening a realm (e.g. file doesn't exist, etc.)
2. those errors are already tested elsewhere